Downsample/Downsample.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+Import a ZDF point cloud and downsample it.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+
+#include <iostream>
+Zivid::PointCloud downsample(const Zivid::PointCloud&, int);
+Eigen::MatrixXi downsampleandround(const Eigen::MatrixXi&, int);
+Eigen::MatrixXf gridsum(const Eigen::MatrixXf&, int);
+Eigen::MatrixXf sumline(const Eigen::MatrixXf&, int);
+float nantozero(float);
+void visualizepointcloud(const Zivid::PointCloud&, Zivid::Application&);
+
+int main()
+{
+	try
+	{
+		Zivid::Application zivid;
+		
+		std::string filename = "Zivid3D.zdf";
+		std::cout << "Reading " << filename << " point cloud" << std::endl;
+		Zivid::Frame frame(filename);
+
+		auto pointCloud = frame.getPointCloud();
+
+		auto downsamplingFactor = 1;
+
+		auto pointCloudDownsampled = downsample(pointCloud, downsamplingFactor);
+
+		visualizepointcloud(pointCloud,zivid);
+		visualizepointcloud(pointCloudDownsampled,zivid);
+	}
+
+	catch (const std::exception &e)
+	{
+		std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+		return EXIT_FAILURE;
+	}
+}
+
+Zivid::PointCloud downsample(const Zivid::PointCloud& pointCloud, int downsamplingFactor)
+{
+	/*
+	Function for downsampling a Zivid point cloud. The downsampling factor represents the denominator of a fraction that represents the size of the downsampled point cloud relative to the original point cloud, e.g. 2 - one-half,  3 - one-third, 4 one-quarter, etc.
+	*/
+
+	if ((pointCloud.height() % downsamplingFactor) || (pointCloud.width() % downsamplingFactor))
+	{
+		throw std::invalid_argument("Downsampling factor (" + std::to_string(downsamplingFactor) + ") has to a factor of the width (" + std::to_string(pointCloud.width()) + ") and height (" + std::to_string(pointCloud.height()) + ") of the input point cloud.");
```

Comment:
What do you mean by splitting it?

Reply:
Just split it into several lines. I ran it through the default `clang-format` which suggests this:
```c++
    throw std::invalid_argument(
        "Downsampling factor (" + std::to_string(downsamplingFactor) +
        ") has to a factor of the width (" +
        std::to_string(pointCloud.width()) + ") and height (" +
        std::to_string(pointCloud.height()) + ") of the input point cloud.");
```
But this would also be ok given the current overall line length in that file:
```c++
    throw std::invalid_argument(
        "Downsampling factor (" + std::to_string(downsamplingFactor) + ") has to a factor of the width (" +
        std::to_string(pointCloud.width()) + ") and height (" + std::to_string(pointCloud.height()) + ") of the input point cloud."
        );
```
This will sort itself out as soon as we add automatic formatting. I'll lobby for doing that sooner rather than later in our upcoming sprint planning.

Reply:
Done.